### ISSUE TITLE
Questions about setting eval mode in DBAL and BALD

### ISSUE THREAD

**Issue by LightersWang** [acl21/deep-active-learning-pytorch#5]
Hi!
Thanks for your solid work on building such a useful tool for active learning in classification.
Now, I'm adapting your code to active learning in medical image classification. 
While reading your code, I encountered questions about setting the clf_model to evaluation mode in DBAL and BALD.

In line 300 - 303 and 364 - 368 of [Sampling.py](https://github.com/acl21/deep-active-learning-pytorch/blob/main/pycls/al/Sampling.py#L300), you first set all the modules in clf_model to train mode and set the batch norm modules to eval mode. Although doing so leaves the dropout layers as train mode, it also sets the conv layers and fc layer as the train mode, too. Is it ok to do this? Or should we turn the model to eval mode first, and then set the dropout layers as train mode?

I'm aware it's a detailed question about the implementation, raising it may seem unnecessary, but I would appreciate hearing from you.
